### ISSUE TITLE
Nullable state not tracked for `ref` variables

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#25868]
A warning is reported for the following:
```c#
class C
{
    static int M(ref string? str)
    {
        if (str == null)
            str = "";
        return str.Length; // CS8602
    }
}
```

**Comment by cston** [comment]
cc @bkoelman

**Comment by cston** [comment]
Similarly for:
```c#
class C
{
    static int M(string? x)
    {
        ref string? y = ref x;
        if (y == null)
            y = "";
        return y.Length; // CS8602
    }
}
```

**Comment by bkoelman** [comment]
Should the fix be in VS2019-Preview3? Because I'm still getting the warning there.

**Comment by jcouv** [comment]
I think the fix made it into VS2019 update 1 preview 1 (dev16.1-preview1, which is not yet publicly available)..
